apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -489,6 +493,35 @@ export default class ProjectManager {
     }
   }
 
+  private async loadHelper(versionId?: string, isPreview?: boolean) {
+    let sources: ProjectSources | undefined;
+    try {
+      sources = await this.sourcesStore.load(this.channelId, versionId);
+      // If we are previewing a project, we don't want to save the source.
+      if (!isPreview) {
+        this.lastSource = JSON.stringify(sources);
+      }
```

Comment:
Checking my understanding, does this assume that the lab will save the sources if `isPreview` is false? Doesn't look like the project manager actually performs a save either way right?

Reply:
Also, are there any usages of `loadHelper` with `isPreview = true` or would that be in a different change?

Reply:
Yeah I added this because we'll need it for the view feature but it isn't used anywhere yet.
I will make this clearer, we aren't ever saving the code. When all is said and done there will be 3 cases for this function:
1. We are loading the most recent sources on level load (what existed before). 
2. We just restored a previous version and are now loading the most recent sources to send back to the lab.
3. We are previewing a specific version (haven't done this end to end yet).

For 1 and 2, the sources have already been saved and we are sending them to the lab. When the lab calls `save`, we want to compare the new code to what we loaded so we don't trigger an unnecessary save (that's the purpose of `lastSource`. For 3 we don't want the project manager to track this code at all, as it's for a preview state.

Reply:
I updated the comments here, lmk if this makes sense!